sycl/test-e2e/format.py

```diff
@@ -154,18 +195,26 @@ def execute(self, test, litConfig):
         if isinstance(script, lit.Test.Result):
             return script
 
-        devices_for_test = self.select_devices_for_test(test)
-        if not devices_for_test:
-            return lit.Test.Result(
-                lit.Test.UNSUPPORTED, "No supported devices to run the test on"
-            )
-
-        substitutions = lit.TestRunner.getDefaultSubstitutions(test, tmpDir, tmpBase)
+        devices_for_test = []
         triples = set()
-        for sycl_device in devices_for_test:
-            (backend, _) = sycl_device.split(":")
-            triples.add(get_triple(test, backend))
+        if "run-mode" not in test.config.available_features:
+            triples = self.select_triples_for_test(test)
+            if not triples:
+                return lit.Test.Result(
+                    lit.Test.UNSUPPORTED, "No supported backend to build for"
+                )
+        else:
+            devices_for_test = self.select_devices_for_test(test)
+            if not devices_for_test:
+                return lit.Test.Result(
+                    lit.Test.UNSUPPORTED, "No supported devices to run the test on"
```

Comment:
have you tried building on one system and running on a different system yet? have you have any issues with different paths to python or something like that?

Reply:
Yes, i ran some experiments to run the split side on a separate machine after running the build mode on another. I did not run into issues relating to python path, mostly issues related to the prs i've opened, and AOT tests.

Reply:
ah ok cool, i expected this part might have insane nonsense to work through but im glad it seems not too bad